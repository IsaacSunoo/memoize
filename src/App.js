import React, { Component } from 'react';
import { BrowserRouter as Router, Route} from 'react-router-dom';
import LandingPage from './components/pages/LandingPage';
import FlashCardContainer from './components/FlashCardContainer';
import QuizPage from './components/pages/QuizPage';
import CodingPage from './components/pages/CodingPage';
import './App.css';

class App extends Component {
  constructor() {
    super();
    this.state = {
      prototypes: []
    }
  }

  componentDidMount() {
    fetch('http://memoize-datasets.herokuapp.com/api/v1/isData')
      .then(response => response.json())
      .then(proto => {
        this.setState({
          prototypes: proto.isData
        })
      })
      .catch(err => {
        throw new Error(err);
      })
  }

  render() {
    return (
      <Router>
        <div>
          <Route exact path='/' component={LandingPage} />
          <Route path='/flashcards' render={() => (
            <React.Fragment>
              <FlashCardContainer prototype={this.state.prototypes} />
            </React.Fragment>
          )} />
          <Route path='/quiz' render={() => (
            <React.Fragment>
              <QuizPage prototype={this.state.prototypes} />
            </React.Fragment>
          )} />
          <Route path='/coding' component={CodingPage} />
        </div>
      </Router>
    )
  }
}

export default App;
