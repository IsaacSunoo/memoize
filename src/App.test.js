import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow } from 'enzyme';

describe('App', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <App />
    )
  });

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should test the default state', () => {
    expect(wrapper.state('prototypes')).toEqual([]);
    expect(wrapper.state('incorrectQuestions')).toEqual([]);
  });

  it('should test the determineSolution', () => {
    expect(wrapper.state('incorrectQuestions')).toEqual([]);
    wrapper.instance().determineSolution([1]);
    expect(wrapper.state('incorrectQuestions')).toEqual([1]);
  })






})

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});
