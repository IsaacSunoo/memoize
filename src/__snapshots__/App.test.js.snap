// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with all data passed in correctly 1`] = `
<BrowserRouter>
  <div>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      path="/flashcards"
      render={[Function]}
    />
    <Route
      path="/quiz"
      render={[Function]}
    />
    <Route
      path="/results"
      render={[Function]}
    />
    <Route
      component={[Function]}
      path="/coding"
    />
  </div>
</BrowserRouter>
`;
